assets/js/components/PostSearcherAutoSuggest.js

```diff
@@ -72,18 +55,25 @@ const PostSearcherAutoSuggest = ( { setCanSubmit, setMatch } ) => {
 		} else {
 			setCanSubmit( false );
 		}
-	}, [ results ] );
+	}, [ results, setCanSubmit, setMatch ] );
 
 	const onInputChange = useCallback( ( event ) => {
 		setCanSubmit( false );
 		setSearchTerm( event.target.value );
-	}, [] );
+	}, [ setCanSubmit ] );
 
 	useEffect( () => {
 		if ( debouncedValue !== '' ) {
-			postSearch( debouncedValue );
+			const request = API.get(
+				'core', 'search', 'post-search',
+				{ query: encodeURIComponent( debouncedValue ) },
+				{ useCache: false },
+			);
+
+			request.then( setResults )
+				.catch( () => setResults( [] ) );
```

Comment:
This is great 👍  My only nitpick here is that the temporary variable is unnecessary and the promise methods can be chained onto the call itself instead, but the result itself is the same of course.